src/app/lab/lab.component.ts

```diff
@@ -26,26 +26,43 @@ import * as _ from 'lodash';
   templateUrl: './lab.component.html'
 })
 export class LabComponent implements OnInit, OnDestroy {
-  name = 'Climate Lab';
 
-  constructor(viewContainerRef: ViewContainerRef,
-              private chartService: ChartService,
+  public apiCities: string = apiHost + "/api/city/?search=:keyword&format=json";
+  public climateModels: ClimateModel[] = [];
+  public scenarios: Scenario[] = [];
+  public project: Project;
+
+  constructor(private chartService: ChartService,
               private climateModelService: ClimateModelService,
               private scenarioService: ScenarioService,
               private projectService: ProjectService) {
-    // necessary to catch application root view container ref. see:
-    // https://valor-software.com/ng2-bootstrap/#/modals
-    this.viewContainerRef = viewContainerRef;
   }
 
-  public project: Project;
-
-  public apiCities: string = apiHost + "/api/city/?search=:keyword&format=json";
+  ngOnInit() {
+    // Initialize a project if it doesn't exist, otherwise just use an existing one
+    // TODO: This logic should be replaced with actual project handling code
+    //       once the feature exists
+    let projects = this.projectService.list();
+    if (projects.length) {
+      this.project = projects[0];
+    } else {
+      this.project = new Project({
+        id: 'test',
+        name: 'Test Project',
+        description: 'Placeholder for projects feature',
+        scenario: {name: defaultScenario, description: ''},
+        city: defaultCity
+      });
+      this.projectService.add(this.project);
+    }
 
-  private chartData: ChartData[];
-  public scenarios: Scenario[];
+    this.getClimateModels();
+    this.getScenarios();
+  }
 
-  public viewContainerRef: ViewContainerRef;
+  ngOnDestroy() {
+    this.projectService.update(this.project);
```

Comment:
For convention, I think we should also unsubscribe from observables here. Memory leak is not a risk here because I think this component lives for the length of the whole app, as exemplified: https://angular.io/docs/ts/latest/cookbook/component-communication.html#!#bidirectional-service